File: gitdot-web/packages/api/src/endpoint/question/list-questions.ts
import { z } from "zod";
import { QuestionResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListQuestionsRequest = z.object({});
export type ListQuestionsRequest = z.infer<typeof ListQuestionsRequest>;

export const ListQuestionsResponse = z.array(QuestionResource);
export type ListQuestionsResponse = z.infer<typeof ListQuestionsResponse>;

export const ListQuestions = {
  path: "/repository/{owner}/{repo}/questions",
  method: "GET",
  request: ListQuestionsRequest,
  response: ListQuestionsResponse,
} as const satisfies Endpoint;
export type ListQuestions = typeof ListQuestions;

File: gitdot-api/src/endpoint/question/list_questions.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::QuestionResource};

pub struct ListQuestions;

impl Endpoint for ListQuestions {
    const PATH: &'static str = "/repository/{owner}/{repo}/questions";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListQuestionsRequest;
    type Response = ListQuestionsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct ListQuestionsRequest;

pub type ListQuestionsResponse = Vec<QuestionResource>;
